exasol_sphinx_github_pages_generator/generate_index.py

```diff
@@ -0,0 +1,157 @@
+    yield
+    os.chdir(old_working_dir)
+
+
+def find_index(target_worktree: Path, source_branch: str) -> Path:
+    """
+    For the given source_branch find the index.html file in its documentation files,
+    and return its path. Aborts if there are more or less han exactly one (1) index.html file.
+    :param target_worktree: Worktree/path all generated project documentation is put into.
+    :param source_branch: Name of the branch the documentation should be searched for.
+    :return: Path pointing at found index.html file
+    """
+    with change_and_restore(target_worktree):
+        index_list = glob.glob(f'{source_branch}/**/index.html', recursive=True)
+        if len(index_list) != 1:
+            sys.exit(inspect.cleandoc(f"""
+                    Your generated documentation does not include the right amount of index.html files (1). 
+                    Instead it includes {len(index_list)} in path {target_worktree}/{source_branch}
+                    """))
+        index_path = index_list[0]
+    return Path(index_path)
+
+
+def generate_release_dicts(release_list: Union[Generator[str, None, Any], List[str]], source_branch: str, target_worktree: Path) \
+        -> List[Dict[str, str]]:
+    """
+    Given a list of releases, generate a list of dictionaries containing the name of the release, and the path to its
+    documentation relative to targen_worktree. The current release is titled "latest".
+    :param release_list: List or Generator of release-names as strings.
+    :param source_branch: The branch we are currently generating the documentation for.
+    :param target_worktree:  Worktree/path all generated project documentation is put into.
+    :return: List of dictionaries containing the release name and path to its index.html file.
+    """
+    release_list_dicts = [{"release": "latest",
+                           "release_path": f"{find_index(target_worktree, source_branch)}"}]
+    for release in release_list:
+        if release == "_sources" or release == "_static":
+            continue
+        if release != source_branch:
+            release_list_dicts.append({"release": release,
+                                       "release_path": f"{find_index(Path('.'), release)}"})
+    return release_list_dicts
+
+
+def get_releases(target_branch: str, target_branch_exists_remote: bool, source_branch: str, target_worktree: Path) \
+        -> List[Dict[str, str]]:
+    """
+    Find all releases already documented in target_branch, and return them tih the path pointing to their index.html
+    file. Additionally also return the current release and index.html-file-path. The current release is titled "latest".
+    Found releases are returned as a list of dictionaries.
+    In order to find the releases in the target_branch, it is checked out in an additional temporary worktree.
+    Takes the root-directory name for each release and returns it. Gives bad results if directory names contain slashes.
```

Comment:
Directory and file names can't contain slashes under Linux. For windows this might be different, but to be honest, this tool will be used mainly in the CI. https://stackoverflow.com/questions/9847288/is-it-possible-to-use-in-a-filename 

However, what happens, when the branches contain slashes, do we get directories and subdirectories?

Reply:
I was talking about branch names. i will rewrite this comment a bit to make i clearer.